server/src/main/java/org/apache/druid/segment/loading/SegmentLoaderLocalCacheManager.java

```diff
@@ -157,7 +184,15 @@ public File getSegmentFiles(DataSegment segment) throws SegmentLoadingException
       try {
         StorageLocation loc = findStorageLocationIfLoaded(segment);
         String storageDir = DataSegmentPusher.getDefaultStorageDir(segment, false);
-
+        if (loc != null) {
+          File localStorageDir = new File(loc.getPath(), storageDir);
+          if (checkSegmentFilesIntact(localStorageDir)) {
```

Comment:
nit: maybe we can push this check inside `findStorageLocationIfLoaded(segment)` itself next to its `dirExists?` check.

Reply:
Done.